poly-commitment/src/srs.rs

```diff
@@ -104,6 +108,21 @@ impl<G: CommitmentCurve> SRS<G> {
             return;
         }
 
+        self.lagrange_bases_cache.as_ref().map(|cache| {
```

Comment:
Similarly here: it's surprising if `add_lagrange_basis` actually does network IO. The cache shouldn't be integrated here